controllers/idler/idler_controller_test.go

```diff
@@ -256,6 +266,95 @@ func TestEnsureIdling(t *testing.T) {
 			})
 		})
 	})
+
+	t.Run("Create notification the first time resources Idled", func(t *testing.T) {
+		// given
+		idler := &toolchainv1alpha1.Idler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "alex-stage",
+				Labels: map[string]string{
+					toolchainv1alpha1.OwnerLabelKey: "alex",
+				},
+			},
+			Spec: toolchainv1alpha1.IdlerSpec{TimeoutSeconds: 60},
+		}
+		namespaces := []string{"dev", "stage"}
+		usernames := []string{"alex"}
+		nsTmplSet := newNSTmplSet(test.MemberOperatorNs, "alex", "advanced", "abcde11", namespaces, usernames)
+		mur := newMUR("alex")
+		reconciler, req, cl, _ := prepareReconcile(t, idler.Name, newGetHostClusterReady, idler, nsTmplSet, mur)
+		idlerTimeoutPlusOneSecondAgo := time.Now().Add(-time.Duration(idler.Spec.TimeoutSeconds+1) * time.Second)
+		preparePayloads(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo)
+
+		// first reconcile to track pods
+		res, err := reconciler.Reconcile(context.TODO(), req)
+		require.NoError(t, err)
+		assert.True(t, res.Requeue)
+		assert.Equal(t, int(res.RequeueAfter), 0)
+
+		// second reconcile should delete pods and create notification
+		res, err = reconciler.Reconcile(context.TODO(), req)
+		//then
+		assert.NoError(t, err)
+		memberoperatortest.AssertThatIdler(t, idler.Name, cl).
+			HasConditions(memberoperatortest.Running(), memberoperatortest.IdlerNotificationCreated())
+		//check the notification is actually created
+		hostCl, _ := reconciler.GetHostCluster()
+		notification := &toolchainv1alpha1.Notification{}
+		err = hostCl.Client.Get(context.TODO(), types.NamespacedName{
+			Namespace: test.HostOperatorNs,
+			Name:      "alex-stage-idled",
+		}, notification)
+		require.NoError(t, err)
+
```

Comment:
We should also add another reconcile to ensure the notification is not recreated/changed

Reply:
I do check in a separate test case that the notification is not recreated - https://github.com/ranakan19/member-operator/blob/a58c63eb12b9f083fe3273d956221f77d8980f9e/controllers/idler/idler_controller_test.go#L541